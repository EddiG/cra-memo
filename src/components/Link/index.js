// @flow
import React from 'react';

type Props = {
  href: string,
  children?: React.Children,
};

const Link = ({ href, children, ...props }: Props) => (
  <a href={href} {...props}>{children}</a>
);

Link.defaultProps = {
  children: null,
};

export default Link;
