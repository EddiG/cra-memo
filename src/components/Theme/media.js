import { css } from 'styled-components';

const baseFontSize = 16;

const sizes = {
  desktop: 1024,
  tablet: 768,
  phone: 376,
};

export default Object.keys(sizes).reduce((acc, target) => {
  const emSize = sizes[target] / baseFontSize;
  acc[target] = (...args) => css`
    @media (max-width: ${emSize}em) {
      ${css(...args)}
    }
  `;
  return acc;
}, {});
