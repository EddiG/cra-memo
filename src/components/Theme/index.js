// @flow
import React from 'react';
import { ThemeProvider } from 'styled-components';
import media from './media';

type Props = {
  children?: React.Children,
};

const theme = {
  colors: {
    fg: {
      main: '#b1b1b1',
      avatar: '#53bdff',
      button: '#fff',
    },
    bg: {
      nav: '#333333',
      main: '#fff',
      button: '#e0b305',
    },
  },
  media,
};

const Theme = ({ children }: Props) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

Theme.defaultProps = {
  children: null,
};

export default Theme;
