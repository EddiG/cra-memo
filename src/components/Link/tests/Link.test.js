import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../index';
import Theme from '../../Theme';

describe('Link component', () => {
  it('should rendered successfully', () => {
    const link = renderer.create(<Theme><Link href="#" /></Theme>).toJSON();
    expect(link).toMatchSnapshot();
  });
});
