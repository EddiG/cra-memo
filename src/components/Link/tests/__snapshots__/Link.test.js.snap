// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should rendered successfully 1`] = `
<a
  href="#"
/>
`;
